bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/storage/ldb/EntryLocationIndexTest.java

```diff
@@ -108,4 +109,35 @@ public void addLedgerAfterDeleteTest() throws Exception {
 
         idx.close();
     }
+
+    // test non exist entry
+    @Test
+    public void testDeleteSpecialEntry() throws IOException {
+        File tmpDir = File.createTempFile("bkTest", ".dir");
+        tmpDir.delete();
+        tmpDir.mkdir();
+        tmpDir.deleteOnExit();
+
+        EntryLocationIndex idx = new EntryLocationIndex(serverConfiguration, KeyValueStorageRocksDB.factory,
+                                                        tmpDir.getAbsolutePath(), NullStatsLogger.INSTANCE);
+
+        // Add some dummy indexes
+        idx.addLocation(40312, -1, 1);
+        idx.addLocation(40313, 10, 2);
+        idx.addLocation(40320, 0, 3);
+
+        // Add more indexes in a different batch
+        idx.addLocation(40313, 11, 5);
+        idx.addLocation(40313, 12, 6);
+        idx.addLocation(40320, 1, 7);
+
+        // delete a non exist entry
+        idx.delete(40312);
```

Comment:
I am a bit confused about this test here

you delete a ledger that exists in the index, no? you already added an entry location of 40312?

Reply:
if i don't add a index, it will break at here : 

```
Entry<byte[], byte[]> firstKeyRes = locationsDb.getCeil(firstKeyWrapper.array);
                if (firstKeyRes == null || ArrayUtil.getLong(firstKeyRes.getKey(), 0) != ledgerId) {
                    // No entries found for ledger
                    if (log.isDebugEnabled()) {
                        log.debug("No entries found for ledger {}", ledgerId);
                    }
                    continue;
                }
```

So, i think it will occurred when entry not exist but ledger exist, so i write `entryid` as `-1L`

Reply:
I  see. 👍